src/metabase/integrations/slack.clj

```diff
@@ -144,27 +181,35 @@
        (filter (complement :deleted))
        (filter (complement :is_bot))))
 
-(defn- files-channel* []
-  (or (channel-with-name files-channel-name)
-      (let [message (str (tru "Slack channel named `metabase_files` is missing!")
-                         " "
-                         (tru "Please create or unarchive the channel in order to complete the Slack integration.")
-                         " "
-                         (tru "The channel is used for storing images that are included in dashboard subscriptions."))]
-        (log/error (u/format-color 'red message))
-        (throw (ex-info message {:status-code 400})))))
-
-(def ^{:arglists '([])} files-channel
-  "Calls Slack api `channels.info` to check whether a channel named #metabase_files exists. If it doesn't, throws an
-  error that advices an admin to create it."
+(def ^:private ^{:arglists '([channel-name])} files-channel*
   ;; If the channel has successfully been created we can cache the information about it from the API response. We need
   ;; this information every time we send out a pulse, but making a call to the `conversations.list` endpoint everytime we
   ;; send a Pulse can result in us seeing 429 (rate limiting) status codes -- see
   ;; https://github.com/metabase/metabase/issues/8967
   ;;
   ;; Of course, if `files-channel*` *fails* (because the channel is not created), this won't get cached; this is what
   ;; we want -- to remind people to create it
-  (memoize/ttl files-channel* :ttl/threshold (u/hours->ms 6)))
+  ;;
+  ;; The memoized function is paramterized by the channel name so that if the name is changed, the cached channel details
+  ;; will be refetched.
+  (memoize/ttl
+   (fn [channel-name]
+       (or (channel-with-name channel-name)
```

Comment:
nit indentation is off here. Also, you allow for nil channel-name. We could skip getting the channel list in that case i guess. `(or (when channel-name (channel-with-name channel-name)) ...)`

Reply:
The `slack-files-channel` setting defaults to `metabase_files` if unset, or if `nil` or an empty string is saved, so this shouldn't ever happen. But I can add that as a safeguard.